assets/src/edit-story/components/keyboard/index.js

```diff
@@ -0,0 +1,164 @@
+ *
+ * @param {{current: Node}} ref
+ * @param {string|Array|Object} keyNameOrSpec
+ * @param {function(KeyboardEvent)} callback
+ * @param {Array|undefined} opts
+ */
+export function useKeyDownEffect(
+  ref,
+  keyNameOrSpec,
+  callback,
+  opts = undefined
+) {
+  const { keys } = useContext(Context);
+  useEffect(
+    () => {
+      const node = ref.current;
+      if (!node) {
+        return undefined;
+      }
+      const mousetrap = getOrCreateMousetrap(node);
+      const keySpec = resolveKeySpec(keys, keyNameOrSpec);
+      const handler = createKeyHandler(keySpec, callback);
+      mousetrap.bind(keySpec.key, handler);
+      return () => {
+        mousetrap.unbind(keySpec.key);
+      };
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    opts || []
+  );
+}
+
+/**
+ * See https://craig.is/killing/mice#keys for the supported key codes.
+ *
+ * @param {string|Array|Object} keyNameOrSpec
+ * @param {function(KeyboardEvent)} callback
+ * @param {Array|undefined} opts
+ */
+export function useGlobalKeyDownEffect(
+  keyNameOrSpec,
+  callback,
+  opts = undefined
+) {
+  if (!globalRef.current) {
+    globalRef.current = document;
+  }
+  useKeyDownEffect(globalRef, keyNameOrSpec, callback, opts);
+}
+
+/**
+ * @param {{current: Node}} ref
+ * @param {Array|undefined} opts
+ */
+export function useEscapeToBlurEffect(ref, opts = undefined) {
+  useKeyDownEffect(
+    ref,
+    { key: 'esc', input: true },
+    () => {
+      const { current } = ref;
+      const { activeElement } = document;
+      if (current.contains(activeElement)) {
+        activeElement.blur();
+      }
+    },
+    opts
+  );
+}
+
+/**
+ * @param {Node} node The DOM node.
+ * @return {Mousetrap} The Mousetrap object that will be used to intercept
+ * the keyboard events on the specified node.
+ */
+function getOrCreateMousetrap(node) {
+  return node[PROP] || (node[PROP] = new Mousetrap(node));
+}
+
+/**
+ * @param {Object} keyDict
+ * @param {string|Object} keyNameOrSpec
+ */
+function resolveKeySpec(keyDict, keyNameOrSpec) {
+  const props =
+    typeof keyNameOrSpec === 'string' || Array.isArray(keyNameOrSpec)
+      ? { key: keyNameOrSpec }
+      : keyNameOrSpec;
+  const {
+    key: keyOrArray,
+    shift = false,
+    repeat = true,
+    input = false,
+  } = props;
+  const mappedKeys = []
+    .concat(keyOrArray)
+    .reduce((keys, key) => keys.concat(keyDict[key] || key), []);
+  const allKeys = addMods(mappedKeys, shift);
+  return { key: allKeys, shift, repeat, input };
+}
+
+function addMods(keys, shift) {
+  if (!shift) {
+    return keys;
+  }
+  return keys.concat(keys.map((key) => `shift+${key}`));
+}
+
+function createKeyHandler({ repeat, input }, callback) {
+  return (evt) => {
+    if ((input || isValidTarget(evt.target)) && (repeat || !evt.repeat)) {
+      callback(evt);
+      // The `false` value instructs Mousetrap to cancel event propagation
+      // and default behavior.
+      return false;
```

Comment:
Yeah. I figured if we want to keep propagation, we will just eventually have the callback to return `false` and we can do `callback(evt) !== false -> stopPropagation()`. But I do think the `stopPropagation` should be a default in such system since one has no way of knowing easily what's further up the DOM tree.